docs/source/daml/intro/3_Data.rst

```diff
@@ -218,7 +218,7 @@ You have already met the type ``ContractId a``, which references a contract of t
   :start-after: -- ID_REF_TEST_BEGIN
   :end-before: -- ID_REF_TEST_END
 
-The scenario above uses the ``fetch`` function, which retrieves the arguments of an active contract using its contract ID.
+The script above uses the ``fetch`` function, which retrieves the arguments of an active contract using its contract ID. ``fetch`` is not directly exposed to ledger clients. Therefore, we create a helper template with the functionality relying on ``fetch`` exposed via choices. We will learn more about choices in :doc:`the next section <4_Transformations>`. We can call those choices by specifying the template and the choice as arguments to ``createAndExerciseCmd``. Note that within the choices we omit the ``cmd`` suffix since that code is executed directly by the ledger instead of building up commands on the client.
```

Comment:
Wouldn't it be better to introduce the query functionality here?

Reply:
The problem with `query` is that you cannot generalize it if you want to combine this with more commands in a single transaction. I’m open to using it here but I think section 4 needs to mention this pattern.

Reply:
It always struck me as odd that I have to do fetch-assert pairs in a submit block anyway. Couldn't we offer a `fetch`-like function on the `Script` Action so that one can just do `args <- fetchContract @T cid; assert(args.foo == bar)` or similar?

Reply:
Fair enough, I guess if the JSON API already provides that endpoint this is reasonable. I still don’t like the fact that I have to do a linear search over the gRPC API but it’s not too bad either. Do you want to stick with `fetchContract` or do you have another name suggestion?

Reply:
Replaced by `queryContractId`.